.circleci/config.yml

```diff
@@ -0,0 +1,137 @@
+version: 2.1
+
+defaults: &defaults
+  working_directory: ~/repo
+  environment:
+    LEIN_ROOT: "true"
+    JVM_OPTS: -Xmx3200m
+
+jobs:
+  checkout_code:
+    docker:
+      - image: circleci/clojure:lein-2.9.0
+    <<: *defaults
+    steps:
+      - checkout
+      - restore_cache:
+          keys:
+            - v2-dependencies-{{ checksum "project.clj" }}
+            - v2-dependencies-
+      - run:
+          name: Downloading deps
+          command: |
+            lein deps
+      - save_cache:
+          paths:
+            - ~/.m2
+          key: v2-dependencies-{{ checksum "project.clj" }}
+      - persist_to_workspace:
+          root: /home/circleci
+          paths:
+            - .m2
+            - repo
+
+  lint_code:
+    docker:
+      - image: circleci/clojure:lein-2.9.0
+    <<: *defaults
+    steps:
+      - attach_workspace:
+          at: /home/circleci
+      - run:
+          name: Install make
+          command: |
+            sudo apt-get install make
+      - run:
+          name: Running Eastwood
+          command: |
+            make eastwood
+      - run:
+          name: Running cljfmt
+          command: |
+            make cljfmt
+
+  test_code:
+    description: Run tests
+    parameters:
+      jdk_version:
+        description: Version of JDK to test against
+        type: string
+        default: ""
+      clojure_version:
+        description: Version of Clojure to test against
+        type: string
+    docker:
+      - image: circleci/clojure:<< parameters.jdk_version >>lein-2.9.0
```

Comment:
Extremely minor nit, but we could include the `-` here rather than having to append it to all the `jdk_version`s below.

Reply:
The relevant docker images are

- `circleci/clojure:lein-2.9.0`
- `circleci/clojure:openjdk-8-lein-2.9.0`
- `circleci/clojure:openjdk-11-lein-2.9.0`

Thus, I'm actually using some string manipulation to pick the JDK here.

For fast builds, it's advisable to use a circleci build image, as it's more likely to be cached. However, we can look for other images/roll our own. But I find it's better to see what we need across a few projects first.

Reply:
Got it, thanks! :+1: